### PR TITLE
ADD: ACINQ wallet support

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by marcosrdz** [comment]
Updated gradients.

**Comment by marcosrdz** [comment]
@Overtorment  whats stopping this from being merged in?

**Comment by Overtorment** [comment]
I want to test it again.
Haven't had the change yet

File: class/app-storage.js [hunk 1]
```diff
@@ -150,6 +151,9 @@ export class AppStorage {
             case HDLegacyBreadwalletWallet.type:
               unserializedWallet = HDLegacyBreadwalletWallet.fromJson(key);
               break;
+            case ACINQStrikeLightningWallet.type:
+              unserializedWallet = ACINQStrikeLightningWallet.fromJson(key);
```
Comment (Overtorment) [inline class/app-storage.js]: init() missing

File: BlueComponents.js [hunk 2]
```diff
@@ -1340,6 +1386,7 @@ export class BlueListTransactionItem extends Component {
         const currentDate = new Date();
         const now = (currentDate.getTime() / 1000) | 0;
         const invoiceExpiration = this.props.item.timestamp + this.props.item.expire_time;
+        console.warn(invoiceExpiration);
```
Comment (Overtorment) [inline BlueComponents.js]: del

File: screen/lnd/lndCreateInvoice.js [hunk 3]
```diff
@@ -109,6 +116,9 @@ export default class LNDCreateInvoice extends Component {
                 />
               </View>
               <BlueDismissKeyboardInputAccessory />
+              <BlueSpacing20 />
+              <BlueTextCentered>{this.state.fromWallet.optionalDisclosureDetail}</BlueTextCentered>
```
Comment (Overtorment) [inline screen/lnd/lndCreateInvoice.js]: for non-strike it just eats 40 px space plus one empty text row? better disable this block for non-strike

File: screen/wallets/import.js [hunk 4]
```diff
@@ -87,6 +88,13 @@ export default class WalletsImport extends Component {
         await lnd.fetchTransactions();
         await lnd.fetchBalance();
         return this._saveWallet(lnd);
+      } else if (text.startsWith('sk_')) {
+        // ACINQ Strike
+        const wallet = new ACINQStrikeLightningWallet();
+        wallet.setSecret(text);
+        await wallet.authenticate();
```
Comment (Overtorment) [inline screen/wallets/import.js]: shouldnt it be in try-catch..?

Comment (Overtorment) [reply screen/wallets/import.js]: also, import should also be implemented in `screen/wallets/scanQrWif.js`

File: class/app-storage.js (hunk 1 above)
Comment (Overtorment) [reply class/app-storage.js]: ok now I see that init is actually in `fromJson()` and not needed to restore acinq from storage

**Comment by Overtorment** [comment]
oh yeah, and the tests are missing. you can write a test to import strike wallet (acinq key is in secure repo), check that it can fetch txs (20 atm), fetch balance (>0), create userinvoice for 1 sat. additionally, pay that invoice with lndhub wallet, then fetch txs and balance again and check that.

**Comment by GladosBlueWallet** [comment]
This was a triumph. I'm making a note here: HUGE SUCCESS.

 [android] https://appetize.io/app/9bqjr2zduntcw0xnjkvzhxg0jw

**Comment by GladosBlueWallet** [comment]
This was a triumph. I'm making a note here: HUGE SUCCESS.

 [android] https://appetize.io/app/dd5mtgcvfwrkc80j4x655xabgm

**Comment by Overtorment** [comment]
cleanup

**Pull request closed without merging**